topdown/tokens_test.go

```diff
@@ -136,3 +152,308 @@ func TestParseTokenHeader(t *testing.T) {
 		}
 	})
 }
+
+func TestEncodeSignVerify(t *testing.T) {
```

Comment:
I might be totally missing something, but it seems like these are exercising the `jws` and `jwk` packages and not really the `builtinJWTEncodeSign` function per-say. IMO i'd rather see the compliance testing for the token signing in the libraries and the builtin unit tests basically just making sure to call the right library API's, like what you have in `topdown_test`, does that make sense?

Reply:
No, you are not missing. Those were(are) there because in the beginning I everything was under vendor. I will remove those tests.

Reply:
As far as the libraries goes, I do not have strong opinions on how it should be incorporated.  I crafted these libraries specifically for OPA. The original thinking was to use Glide and vendor them.

Having said that, there is at least one other OPA outstanding feature requests related to JWX that will require more changes to the library.  Not sure this will push your opinion one way or another.

https://github.com/open-policy-agent/opa/issues/1131

I only plan to tackle this one after I am done with LDAP support

Reply:
Added the library to glide and update yaml. Just FYI, unfortunately a "glide update" updates the entire vendors and touches a million files that git adds. Then you have to revert everything with exception of the actual directory you need.

Reply:
@repenno I'm expecting the headers, payload, and JWK values to be passed as normal values (e.g., objects) not JSON serialized strings.

Reply:
Unfortunately we'll have to take a full glide update, as-is the lock file is updated but the files aren't in sync. This might cause problems for projects embedding OPA.

Shorter term we could/should maybe look at using dep until the module stuff is more fully baked.